.github/workflows/style.yml

```diff
@@ -22,9 +22,11 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: '3.10'
+          cache: 'pip'
+          cache-dependency-path: ./api/requirements-dev.txt
 
       - name: Python dependencies
-        run: pip install ruff
+        run: pip install -r ./api/requirements-dev.txt
```

Comment:
I think you only need to install ruff here.

Reply:
The general benefits for putting `ruff` in  `requirements-dev.txt` here include:
1. pin the required packages including linters in `requirements-dev.txt` as well
2. `requirements-dev.txt` is designed for general development purpose, covering testing, lining and etc.
3. `requirements-dev.txt` helps to cache the pip packages in Github actions for testing and lining, saving execution time and the cost.

Reply:
I think the style check task here only relies on the ruff package, the others are redundant. And I believe that only installing the ruff package and using action cache doesn't make much of a difference in terms of time consumption. 

P.S. I think we should follow the Pragmatic Programming principle, avoid blindly following established norms, and hope that each piece of code has its most reasonable explanation in decision-making.

Reply:
No problem. Will fix it and follow your suggestions.

Reply:
Have fixed the problem, by following your advice.